### PR TITLE
Release build error fix + optimization

### PR DESCRIPTION
**Pull request by Lakshya2610**
* Wrapped some dev only code with UNITY_EDITOR macro that was causing error when building for release
* Converted `LookAtObject` in `PlayerController` to coroutine to update at a custom rate instead of every frame
	- This is to improve performance as raycast every frame is taxing on CPU

### CODE REVIEW

**Review by dargonfyl — APPROVED** [review]

**Review by ZuhabWasim — APPROVED** [review]
Looks good, let's merge this after mom's closet is merged to main though to avoid any issues.

**Pull request merged**
